azkaban-executor/azkaban-emr-jobtype/src/main/java/uk/gov/dwp/dataworks/azkaban/jobtype/EMRStep.java

```diff
@@ -527,12 +550,12 @@ private void configureCluster(AmazonElasticMapReduce emr, String clusterId) thro
             // We may still be initializing, so wait until we are waiting for steps.
             // Potentially maybe other executors listening, so check step concurrency also.
             while (!clusterDetails.getCluster().getStatus().getState().equals("WAITING")) {
-                String current_state = clusterDetails.getCluster().getStatus().getState();
-                info("Waiting for state WAITING...current state is " + current_state);
+                String currentState = clusterDetails.getCluster().getStatus().getState();
+                info("Waiting for state WAITING...current state is " + currentState);
 
-                if (current_state.equals("TERMINATED")) {
-                    info("Exiting with error as current state is " + current_state);
-                    throw new RuntimeException("Cluster has been terminated");
+                if (currentState.contains("TERMINATED") || currentState.equals("TERMINATING")) {
```

Comment:
Not tempted to do contains 'TERMINAT'? ;-)

Reply:
Ha ha yes I was actually